src/osd/OSD.cc

```diff
@@ -7458,6 +7467,13 @@ void OSD::_committed_osd_maps(epoch_t first, epoch_t last, MOSDMap *m)
 	do_restart = true;
       }
     }
+    if (osdmap->test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS) !=
+	newmap->test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS)) {
+      dout(10) << __func__ << " REQUIRE_LUMINOUS flag changed in " << newmap->get_epoch()
+	       << dendl;
+      clear_pg_stat_queue();
```

Comment:
i'm assuming that the pg enqueuing itself is based on PG::get_osdmap(), which is often an older man than the OSD's OSDMap.

Perhaps what we need here is an OSDService::get_min_pg_epoch() before/after comparison and when *that* crosses the flag threshold then we clear the queue?